src/linking/linkModule.ml

```diff
@@ -801,25 +833,29 @@ let link (em1 : extended_module) libname (em2 : extended_module) =
   let lib_heap_start = align_i32 dylink.memory_alignment old_heap_start in
   let new_heap_start = align_i32 8l (Int32.add lib_heap_start dylink.memory_size) in
 
-  (* The RTS data segments must fit below 4.5MB according to the persistent heap layout. 
-     The first 4MB are reserved for the Rust call stack such that RTS data segments are limited to 512KB. *)
-  let max_rts_stack_size = 4 * 1024 * 1024 in
-  let max_rts_data_segment_size = 512 * 1024 in
-  (if (Int32.to_int new_heap_start) > max_rts_stack_size + max_rts_data_segment_size then
-    (raise (TooLargeDataSegments (Printf.sprintf "The Wasm data segment size exceeds the supported maxmimum of %nMB." max_rts_data_segment_size)))
-  else
-    ()
-  );
+  if uses_memory64 em1.module_ then
+  begin
+    (* The RTS data segments must fit below 4.5MB according to the persistent heap layout. 
```

Comment:
I was worried that 512KB might not be enough if we decide to support unicode and need more static data. But QuickJS seems to do this with just 40KB of additional data so perhaps this is enough.